### PR TITLE
Added request header

### PR DESCRIPTION
**Pull request by JoeStead**
* Migrated to csproj
* Added request method to output

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,8 +1,5 @@
 {
     "projects": [
         "src"
-    ],
```
Comment (jchannon) [inline global.json]: delete the whole file

File: src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (jchannon) [inline src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj]: 1.1

File: src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <AssemblyName>TracingMiddleware.Demo</AssemblyName>
+    <OutputType>Exe</OutputType>
+    <PackageId>TracingMiddleware.Demo</PackageId>
+    <RuntimeFrameworkVersion>1.0.4</RuntimeFrameworkVersion>
```
Comment (jchannon) [inline src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj]: delete line

File: src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../TracingMiddleware/TracingMiddleware.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Nancy" Version="2.0.0-barneyrubble" />
```
Comment (jchannon) [inline src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj]: clinteastwood

File: src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="1.0.2" />
```
Comment (jchannon) [inline src/TracingMiddleware.Demo/TracingMiddleware.Demo.csproj]: 1.1.2 maybe will need to make sure latest

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: 1.1

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    <AssemblyName>TracingMiddleware.Tests</AssemblyName>
+    <PackageId>TracingMiddleware.Tests</PackageId>
+    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
+    <PackageTargetFallback>$(PackageTargetFallback);dotnet;portable-net451+win8</PackageTargetFallback>
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: can probably delete that

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    <RuntimeFrameworkVersion>1.0.4</RuntimeFrameworkVersion>
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: delete

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="../TracingMiddleware/TracingMiddleware.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0-preview-20170106-08" />
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: can be 15.0.0

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0-beta5-build1225" />
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: 2.3.0-beta3-build3705

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageReference Include="Microsoft.NETCore.Platforms" Version="1.0.2" />
+    <PackageReference Include="xunit" Version="2.2.0-beta5-build3474" />
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: 2.3.0-beta3-build3705

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: 1.1

File: src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="1.0.2" />
```
Comment (jchannon) [inline src/TracingMiddleware.Tests/TracingMiddleware.Tests.csproj]: 1.1.2

File: src/TracingMiddleware/TracingMiddleware.csproj [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+v1.0.7 - Guid.TryParse added
+v1.0.6 - ConfigureAwait added
+v1.0.5 - CoreCLR compatible.
+v1.0.4 - Checks environment RequestId is not empty.
+v1.0.3 - Added request path to Request Start/End log entries.
+v1.0.2 - Added tracing for exceptions thrown by middleware.
+v1.0.1 - Added ability to add filters on Default options</PackageReleaseNotes>
+    <PackageProjectUrl>https://github.com/VQComms/TracingMiddleware/</PackageProjectUrl>
+    <PackageLicenseUrl>https://github.com/VQComms/TracingMiddleware/blob/master/LICENSE</PackageLicenseUrl>
+    <NetStandardImplicitPackageVersion>1.6.0</NetStandardImplicitPackageVersion>
```
Comment (jchannon) [inline src/TracingMiddleware/TracingMiddleware.csproj]: delete

File: src/TracingMiddleware/TracingMiddlewareOptions.cs [hunk 15]
```diff
@@ -102,32 +101,20 @@ public TracingMiddlewareOptions(Trace trace, MessageFormat messageFormat, TypeFo
             if (messageFormat == null) throw new ArgumentNullException("messageFormat");
             if (defaultTypeFormat == null) throw new ArgumentNullException("defaultTypeFormat");
 
-            this.trace = trace;
-            this.messageFormat = messageFormat;
+            this.Trace = trace;
+            this.MessageFormat = messageFormat;
             this.defaultTypeFormat = defaultTypeFormat;
             this.isEnabled = isEnabled ?? (() => true);
             this.filters.AddRange(filters);
```
Comment (jchannon) [inline src/TracingMiddleware/TracingMiddlewareOptions.cs]: Could this.Filters be assigned to directly?

Comment (JoeStead) [inline src/TracingMiddleware/TracingMiddlewareOptions.cs]: Nope, because it has some defaults already in there.

**Pull request merged**